how-to/common/public/style/style-changed-preload.js

```diff
@@ -0,0 +1,55 @@
+function setColorScheme(schemeType, darkScheme, lightScheme) {
+	if (schemeType === 'light') {
+		document.body.classList.remove(darkScheme);
+		document.body.classList.add(lightScheme);
+	} else {
+		document.body.classList.remove(lightScheme);
+		document.body.classList.add(darkScheme);
+	}
+}
+
+function kebabCase(input) {
+	if (typeof input === 'string' && input.length > 0) {
+		return (
+			input
+				.replace(/([A-Z])/g, ' $1')
+				.trim()
+				.match(/[^\u0000-\u002F\u003A-\u0040\u005B-\u0060\u007B-\u007F]+/g) ?? []
+		)
+			.join('-')
+			.toLowerCase();
+	}
+	return '';
+}
+
+function createCssVars(prefix, scheme, palette) {
+	const root = document.querySelector(':root');
+	if (root) {
+		root.style.setProperty(`--${prefix}-scheme`, scheme);
+		const keys = Object.keys(palette);
+		for (const key of keys) {
+			root.style.setProperty(`--${prefix}-${kebabCase(key)}`, palette[key]);
+		}
+	}
+}
+
+document.addEventListener('DOMContentLoaded', async () => {
+	try {
+		console.log('Style Change Preload activated');
+
+		const appSessionContextGroup = await fin.me.interop.joinSessionContextGroup('platform-app-channel');
```

Comment:
Confirmed with Michael that he is ok with "a/b/c" instead of "a-b-c" (this gives us flexibility where a part may be two words e.g. "a-b/c" and you couldn't do that with "a-b-c". So platform/events (or something similar but using the topic structure please).

Reply:
Updated